report/node.go

```diff
@@ -182,8 +184,26 @@ func (n Node) WithChildren(children NodeSet) Node {
 
 // WithChild returns a fresh copy of n, with one child merged in.
 func (n Node) WithChild(child Node) Node {
-	n.Children = n.Children.Merge(MakeNodeSet(child))
-	return n
+	result := n.Copy()
+	result.Children = result.Children.Merge(MakeNodeSet(child))
+	return result
+}
+
+// Copy returns a value copy of the Node.
+func (n Node) Copy() Node {
```

Comment:
Something wen wrong, but at the and is that this commit https://github.com/weaveworks/scope/pull/1837/commits/74e564661037cf77247ef8f00ffd9b54be9518c7 undoes https://github.com/weaveworks/scope/commit/1091225bf67b1afa1f1d2b770b52a5b65c85cbe3.

So if I remove it and rebase it should be ok. @rade thoughts?